rs/types/types/src/messages.rs

```diff
@@ -763,27 +763,19 @@ mod tests {
 
     #[test]
     fn serialize_request_via_bincode() {
-        for metadata in [
-            None,
-            Some(RequestMetadata::new(
```

Comment:
The case of `None` here caused a hard to track bug which is why this test was introduced. The version for responses below was just for completeness. I'd say it doesn't hurt to keep it around though.